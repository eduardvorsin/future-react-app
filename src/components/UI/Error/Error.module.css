.error {
	padding: 15px;
	max-width: 600px;
	background-color: var(--error-color-200);
	color: var(--error-color-100);
	border-radius: 4px;
	transition:
		background-color var(--transition-time) var(--transition-easing);
}

.error--center {
	position: absolute;
	top: 50%;
	left: 50%;
	transform: translate(-50%, -50%);
	width: 100%;
}

.error__icon-wrapper {
	margin: 0 auto 15px;
	width: 60px;
	height: 60px;
}

.error__title {
	margin: 0;
	line-height: 1.3;
	text-align: center;
	font-size: 18px;
	color: var(--error-color);
}

@media(min-width:577px) {
	.error {
		padding: 20px;
	}

	.error__title {
		font-size: 20px;
	}

	.error__icon-wrapper {
		width: 80px;
		height: 80px;
		margin: 0 auto 25px;
	}
}

[data-theme='dark'] .error {
	color: var(--light-color-200);
	border-color: var(--error-color-200);
	transition:
		background-color var(--transition-time) var(--transition-easing);
}