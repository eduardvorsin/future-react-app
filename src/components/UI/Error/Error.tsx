import React from 'react';
import Title from '../Title/Title';
import classes from './Error.module.css';

type ErrorProps = {
  className?: string,
  message: string,
  icon?: React.ReactNode,
  isCentered?: boolean,
};

const Error: React.FC<ErrorProps> = ({
  className,
  message,
  icon,
  isCentered,
}) => {
  const errorClasses = [
    className,
    classes.error,
    isCentered ? classes['error--center'] : '',
  ].join(' ');

  return (
    <div
      className={errorClasses}
      role="alert"
    >
      {icon && (
        <div
          className={classes['error__icon-wrapper']}
        >
          {icon}
        </div>
      )}
      <Title
        level={5}
        component='h4'
        variant='error'
        className={classes.error__title}
      >
        {message}
      </Title>
    </div>
  );
};

export default Error;
