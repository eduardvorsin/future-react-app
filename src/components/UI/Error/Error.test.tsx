import React from 'react';
import { render, screen } from '@testing-library/react';
import Error from './Error';
import SunIcon from '../../../assets/images/icons/sun.svg';

describe('Error tests', () => {
  it('is rendered correctly', () => {
    render(
      <Error
        testId='test-error'
        message='test error message'
      />,
    );

    expect(screen.getByTestId<HTMLDivElement>(/test-error/i)).toBeInTheDocument();
    expect(screen.getByText<HTMLHeadingElement>(/test error message/i)).toBeInTheDocument();
  });

  it('if the isCentered prop is passed, the error--center class is added', () => {
    render(
      <Error
        isCentered
        testId='test-error'
        message='test error message'
      />,
    );

    expect(screen.getByTestId<HTMLDivElement>(/test-error/i)).toHaveClass('error--center');
  });

  it('if the icon prop is passed, then the icon is rendered', () => {
    render(
      <Error
        testId='test-error'
        message='test error message'
        icon={<SunIcon title='test-error-icon' />}
      />,
    );

    expect(screen.getByTitle<HTMLElement>(/test-error-icon/i)).toBeInTheDocument();
  });

  it('is a basic snapshot', () => {
    render(
      <Error
        testId='test-error'
        message='test error message'
      />,
    );

    expect(screen.getByTestId<HTMLDivElement>(/test-error/i)).toMatchSnapshot();
  });

  it('is a snapshot with an icon', () => {
    render(
      <Error
        testId='test-error'
        message='test error message'
        icon={<SunIcon title='test-error-icon' />}
      />,
    );

    expect(screen.getByTestId<HTMLDivElement>(/test-error/i)).toMatchSnapshot();
  });
});
