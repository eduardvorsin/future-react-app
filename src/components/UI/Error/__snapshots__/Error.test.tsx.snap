// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error tests is a basic snapshot 1`] = `
<div
  class=" error "
  data-testid="test-error"
  role="alert"
>
  <div
    class="error__icon-wrapper"
  />
  <h4
    class="title error__title title--dark title--level-5"
  >
    test error message
  </h4>
</div>
`;

exports[`Error tests is a snapshot with an icon 1`] = `
<div
  class=" error "
  data-testid="test-error"
  role="alert"
>
  <div
    class="error__icon-wrapper"
  >
    <svgicon
      title="test-error-icon"
    />
  </div>
  <h4
    class="title error__title title--dark title--level-5"
  >
    test error message
  </h4>
</div>
`;
